plugins/debuginfo-install.py

```diff
@@ -74,59 +76,170 @@ def configure(self):
+                continue
 
-        if not self.dbgdone:
-            logger.info(_("No debuginfo packages available to install"))
+            # Remove debuginfo packages if their base packages are in the query.
+            # They can get there through globs and they break the installation
+            # of debug packages with the same version as the installed base
+            # packages. If the base package of a debuginfo package is not in
+            # the query, the user specified a debug package on the command
+            # line. We don't want to ignore those, so we will install them.
+            # But, in this case the version will not be matched to the
+            # installed version of the base package, as that would require
+            # another query and is further complicated if the user specifies a
+            # version themselves etc.
+            for name in list(package_dict.keys()):
+                if name.endswith(Package.DEBUGINFO_SUFFIX):
+                    if name[:-debuginfo_suffix_len] in package_dict:
+                        package_dict.pop(name)
+                if name.endswith(Package.DEBUGSOURCE_SUFFIX):
+                    if name[:-debugsource_suffix_len] in package_dict:
+                        package_dict.pop(name)
+
+            # attempt to install debuginfo and debugsource for the highest
+            # listed version of the package (in case the package is installed,
+            # only the installed version is listed)
+            for packages in package_dict.values():
+                first_package = packages[0]
+
+                # for packages from system (installed) there can be more
+                # packages with different architectures listed and we want to
+                # install debuginfo for all of them
+                if first_package._from_system:
+                    # we need to split them by architectures and install the
+                    # latest version for each architecture
+                    arch_dict = {}
+
+                    for package in packages:
+                        arch_dict.setdefault(package.arch, []).append(package)
+
+                    for package_arch_list in arch_dict.values():
+                        package = package_arch_list[0]
+
+                        if not self._install_debug_from_system(package.debug_name, package):
+                            self.installed_debuginfo_missing.add(str(package))
+
+                        if not self._install_debug_from_system(package.debugsource_name, package):
+                            self.installed_debugsource_missing.add(str(package))
+
+                    continue
+
+                # if the package in question is -debuginfo or -debugsource, install it directly
+                if first_package.name.endswith(Package.DEBUGINFO_SUFFIX) \
+                        or first_package.name.endswith(Package.DEBUGSOURCE_SUFFIX):
+
+                    self._install((first_package, ))
```

Comment:
I would recommend to use all packages that we got like self._install(packages). To keep more options to solver is beneficial, because user could use arguments that have only limited overlap. Like `foo-debuginfo` and `foo-debuginfo-1-5`. Using all packages from the query is not harmful.

Reply:
@lukash Please could you consider the proposed change?

Reply:
Done.